subsys/bluetooth/controller/ll_sw/ull_scan.c

```diff
@@ -639,12 +639,21 @@ uint8_t ull_scan_disable(uint8_t handle, struct ll_scan_set *scan)
 
 		aux_scan = HDR_LLL2ULL(aux_scan_lll);
 		if (aux_scan == scan) {
+			void *parent;
+
 			err = ull_scan_aux_stop(aux);
 			if (err && (err != -EALREADY)) {
 				return BT_HCI_ERR_CMD_DISALLOWED;
 			}
 
-			LL_ASSERT(!aux->parent);
+			/* Use a local variable to assert on auxiliary context's
+			 * release.
+			 * Under race condition a released aux context can be
+			 * allocated for reception of chain PDU of a periodic
+			 * sync role.
+			 */
+			parent = aux->parent;
+			LL_ASSERT(!parent || (parent != aux_scan_lll));
```

Comment:
Correct me if my understanding is wrong.
The part of assert condition `parent != aux_scan_lll` makes sure the auxiliary context was not re-used for other `scan` between call to `ull_scan_aux_stop(aux)` in line 644 and this assert, correct?
If that is true, are you sure it should assert here?

Reply:
Ah yes. Thanks for detailed explanation.